### PR TITLE
refactor: use model and deletion of associated todo

### PR DESCRIPTION
**Pull request by BastienBenard**
## Describe your changes
Use of JiraTicket and PivotTicket models + bug fix on JiraTicket json formatting

## Checklist before requesting a review (magic string, indentation, comment/documentation...)

- [ ] I have detailed the tests to do in my feature/fix in order to prevent consequents regressions (must specify in **Checklist tests**)
- [x] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation (API Doc etc...) - (must specify in **Description** for target version)
- [ ] If it is a consequent feature, I have added thorough tests.
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been added to this project (must specify in **Description**)

### CODE REVIEW

File: src/main/java/fr/openent/supportpivot/model/endpoint/jira/JiraEndpoint.java [hunk 1]
```diff
@@ -118,14 +119,14 @@ private Future<List<PivotTicket>> processSearchResponse(HttpClientResponse respo
                     JiraSearch jiraSearch = new JiraSearch(new JsonObject(body));
                     List<Future<PivotTicket>> futures = new ArrayList<>();
                     jiraSearch.getIssues().forEach(issue -> {
-                        Future<PivotTicket> future = Future.future();
-                        futures.add(future);
+                        Promise<PivotTicket> promisePivotTicket = Promise.promise();
+                        futures.add(promisePivotTicket.future());
```
Comment (BastienBenard) [inline src/main/java/fr/openent/supportpivot/model/endpoint/jira/JiraEndpoint.java:123]: Utilisation de promise car future.complete est déprécié

File: src/main/java/fr/openent/supportpivot/model/endpoint/jira/JiraEndpoint.java [hunk 2]
```diff
@@ -302,53 +298,52 @@ public Future<JsonObject> convertJiraReponseToJsonPivot(final JiraTicket jiraTic
...
         if (fields == null) {
-            promise.complete(jsonPivot);
+            promise.complete(pivotTicket);
         } else {
             String creatorField = jiraCustomFields.getOrDefault(Field.CREATOR, "");
             String creationField = jiraCustomFields.getOrDefault(Field.CREATION, "");
             String uaiField = jiraCustomFields.getOrDefault(Field.UAI, "");
             String idEntField = jiraCustomFields.getOrDefault(Field.ID_ENT, "");
             String statusEntField = jiraCustomFields.getOrDefault(Field.STATUS_ENT, "");
-            String resolutionIws = jiraCustomFields.getOrDefault(Field.RESOLUTION_IWS, "");
```
Comment (BastienBenard) [inline src/main/java/fr/openent/supportpivot/model/endpoint/jira/JiraEndpoint.java:318]: IWS n'est plus utilisé

File: src/main/java/fr/openent/supportpivot/services/JiraServiceImpl.java [hunk 3]
```diff
@@ -198,7 +199,7 @@ private Future<PivotTicket> createJiraTicket(PivotTicket pivotTicket) {
                         if (response.statusCode() == HTTP_STATUS_201_CREATED) {
                             response.bodyHandler(body -> {
                                 JiraTicket jiraTicket = new JiraTicket(new JsonObject(body));
-                                pivotTicket.setIdJira(jiraTicket.getId());
+                                pivotTicket.setIdJira(jiraTicket.getKey());
```
Comment (BastienBenard) [inline src/main/java/fr/openent/supportpivot/services/JiraServiceImpl.java:202]: l'id Jira utiliser par le reste de support correspond a ça key et non son id interne

File: src/main/java/fr/openent/supportpivot/model/jira/JiraFields.java [hunk 4]
```diff
@@ -53,7 +53,9 @@ public JiraFields() {
 
     @Override
     public JsonObject toJson() {
-        return IModelHelper.toJson(this, true, false);
+        JsonObject jsonObject = IModelHelper.toJson(this, true, false);
+        customFields.forEach(jsonObject::put);
+        return jsonObject;
```
Comment (BastienBenard) [inline src/main/java/fr/openent/supportpivot/model/jira/JiraFields.java:58]: Oublie des anciens ticket. IModelHelper#toJson n'ajoute pas automatiquement les object de type Map

File: src/main/java/fr/openent/supportpivot/services/JiraServiceImpl.java [hunk 5]
```diff
@@ -439,83 +440,68 @@ private Future<Void> sendJiraPJ(final String jiraTicketId, final PivotPJ pj) {
...
+                    modifyTicketRequest.setChunked(true)
+                            .write(this.ticketPrepareForUpdate(pivotTicket).toJson().encode());
+
+                    terminateRequest(modifyTicketRequest);
+                })
+                .onFailure(error -> {
+                    LOGGER.error(String.format("[SupportPivot@%s::updateJiraTicket] Error when gathering Jira ticket information : %s",
+                            this.getClass().getSimpleName(), error.getMessage()));
+                    promise.fail(error);
+                });
```
Comment (BastienBenard) [inline src/main/java/fr/openent/supportpivot/services/JiraServiceImpl.java:491]: Utilisation de getFromJira

File: src/main/java/fr/openent/supportpivot/model/pivot/PivotTicket.java [hunk 6]
```diff
@@ -29,7 +29,7 @@ public class PivotTicket implements IModel<PivotTicket>, Cloneable{
     private String uai;
     private String description;
     private String priorite;
-    private List<String> module;
+    private List<String> modules;
```
Comment (BastienBenard) [inline src/main/java/fr/openent/supportpivot/model/pivot/PivotTicket.java:32]: Correction de fautes faites dans les autres ticket

**Review by qpic-cgi — APPROVED** [review]

**Pull request merged**